mozregression/telemetry.py

```diff
@@ -14,10 +17,58 @@
 METRICS = load_metrics(resource_filename(__name__, "metrics.yaml"))
 
 UsageMetrics = namedtuple(
-    "UsageMetrics", ["variant", "appname", "build_type", "good", "bad", "launch"]
+    "UsageMetrics",
+    [
+        "variant",
+        "appname",
+        "build_type",
+        "good",
+        "bad",
+        "launch",
+        "windows_version",
+        "mac_version",
+        "linux_version",
+        "linux_distro",
+        "python_version",
+    ],
 )
 
 
+def get_system_info():
+    """Return a dictionary with various information about the system."""
+    info = {
+        "windows_version": None,
+        "mac_version": None,
+        "linux_version": None,
+        "linux_distro": None,
+        "python_version": None,
+    }
+
+    if mozinfo.os == "mac":
+        try:
+            # Fetch the "release" from tuple containing macOS version information.
+            # See https://docs.python.org/3/library/platform.html#macos-platform.
+            info["mac_version"] = platform.mac_ver()[0]
+        except (AttributeError, IndexError):
+            pass
```

Comment:
Question: could we pass a value like "unknown version" instead of just leaving the field blank here? It would be an annoying bug to track down if telemetry wasn't gathered for a platform and we didn't know why.